### PR TITLE
Introduce pre-commit and ruff

### PR DESCRIPTION
**Pull request by amureki**
The first commit introduces ruff (replacing bandit, flake8, pydocstyle and isort).
The second one fixes issues that ruff discovered.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -70,13 +75,29 @@ source = ["citext"]
 [tool.coverage.report]
 show_missing = true
 
-[tool.isort]
-atomic = true
-line_length = 88
-known_first_party = "citext, tests"
-include_trailing_comma = true
-default_section = "THIRDPARTY"
```
Comment (codingjoe) [inline pyproject.toml:78]: This is missing now, which is why ruff changed the imports.

File: tests/testapp/settings.py [hunk 2]
```diff
@@ -1,5 +1,4 @@
-"""
-Django settings for testapp project.
+"""Django settings for testapp project.
```
Comment (codingjoe) [inline tests/testapp/settings.py:1]: I personally don't love this.
It doesn't improve readability, rather the contrary.

Docstrings are for humans, not machines. I'd rather have a trailing newline, than a dangling typo.

**Review by codingjoe — CHANGES_REQUESTED** [review]
The current style has been chosen with care. I don't mind introducing new tooling, but if you want to introduce style changes, they should be considered separately.

**Comment by amureki** [comment]
@codingjoe fair points, I adjusted the code according to your comments.

**Review by codingjoe — APPROVED** [review]

**Pull request merged**